C error when passing interface method to struct field

### Describe the bug

A struct can have a field of type function:

```v
struct Structure {
    action fn()
}
```

An interface can have a method:

```v
interface Interfacer {
    id int
}
fn (i Interfacer) process() {
    println('proc #${id}')
}
```

But this causes a C error:

```v
struct Process {
    id int
}
i := Interfacer(Process{22})
s := Structure{
    action: i.process
}
```

### Reproduction Steps

Run the following program:

```v
interface Doer{
	job string
}

fn (d Doer) do() {
	println('did:  ${d.job}')
}


struct  Expector {
	do_fn fn() @[required]
}

struct Actioner {
	job string
}

fn main() {
	a := &Actioner{"dier"}
	e := Expector{
		do_fn: Doer(a).do
	}
	e.do_fn()
}
```

### Expected Behavior

Expected console output:

```txt
did: dier
```

### Current Behavior

Instead, an error occurs, something like the following:

```bash
================== C compilation error (from cc): ==============
cc: /usr/tmp/v_10230/main.01JYX7BF26C2Q81F135BK1QYBA.tmp.c:1938:2: error: use of undeclared identifier 'main__Doer_name_table'
cc:  1938 |         main__Doer_name_table[a0->_typ]._method_do(a0);
cc:       |         ^
cc: 1 error generated.
================================================================
```

### Possible Solution

Workaround by @spytheman :

```txt
do_fn: fn [a] () { Doer(a).do() }
```

### Additional Information/Context

_No response_

### V version

V 0.4.11 bd465b5

### Environment details (OS name and version, etc.)

I ran this on a laptop, but had the same error running on my phone:

```bash
|V full version      |V 0.4.11 bd465b5254c10ac2ef2d214bc7845e5cebfef946
|:-------------------|:-------------------
|OS                  |termux, 4.19.191-g6f453d21fb7f-dirty, #1 SMP PREEMPT Fri Feb 23 18:16:46 CST 2024
|Processor           |8 cpus, 64bit, little endian
|Memory              |N/A
|                    |
|V executable        |/data/data/com.termux/files/home/tmp/v/v
|V last modified time|2025-06-28 12:46:57
|                    |
|V home dir          |OK, value: /data/data/com.termux/files/home/tmp/v
|VMODULES            |OK, value: /data/data/com.termux/files/home/.vmodules
|VTMP                |OK, value: /data/data/com.termux/files/usr/tmp/v_10230
|Current working dir |OK, value: /data/data/com.termux/files/home/tmp
|                    |
|Git version         |git version 2.50.0
|V git status        |bd465b52
|.git/config present |true
|                    |
|cc version          |clang version 20.1.7
|gcc version         |clang version 20.1.7
|clang version       |clang version 20.1.7
|tcc version         |N/A
|tcc git status      |thirdparty-unknown-unknown de82a130
|emcc version        |N/A
|glibc version       |N/A
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4289,7 +4289,7 @@ fn (mut g Gen) selector_expr(node ast.SelectorExpr) {
 					left_cc_type := g.cc_type(g.table.unaliased_type(node.expr_type),
 						false)
 					left_type_name := util.no_dots(left_cc_type)
-					g.write('${c_name(left_type_name)}_name_table[${node.expr.name}${g.dot_or_ptr(node.expr_type)}_typ]/**/._method_${m.name}')
+					g.write('${c_name(left_type_name)}_name_table[${node.expr.name}${g.dot_or_ptr(node.expr_type)}_typ]._method_${m.name}')
 				} else {
 					g.write('${g.styp(node.expr_type.idx_type())}_${m.name}')
 				}

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4289,7 +4289,7 @@ fn (mut g Gen) selector_expr(node ast.SelectorExpr) {
 					left_cc_type := g.cc_type(g.table.unaliased_type(node.expr_type),
 						false)
 					left_type_name := util.no_dots(left_cc_type)
-					g.write('${c_name(left_type_name)}_name_table[${node.expr.name}${g.dot_or_ptr(node.expr_type)}_typ]._method_${m.name}')
+					g.write('${c_name(left_type_name)}_name_table[${node.expr.name}${g.dot_or_ptr(node.expr_type)}_typ]/**/._method_${m.name}')
 				} else {
 					g.write('${g.styp(node.expr_type.idx_type())}_${m.name}')
 				}
@@ -4473,7 +4473,7 @@ fn (mut g Gen) gen_closure_fn(expr_styp string, m ast.Fn, name string) {
 			method_name = g.generic_fn_name(rec_sym.info.concrete_types, m.name)
 		}
 	}
-	if rec_sym.info is ast.Interface {
+	if rec_sym.info is ast.Interface && rec_sym.info.get_methods().contains(method_name) {
 		left_cc_type := g.cc_type(g.table.unaliased_type(receiver.typ), false)
 		left_type_name := util.no_dots(left_cc_type)
 		sb.write_string('${c_name(left_type_name)}_name_table[a0->_typ]._method_${method_name}(')

--- a/vlib/v/tests/interfaces/interface_method_closure_test.v
+++ b/vlib/v/tests/interfaces/interface_method_closure_test.v
@@ -1,0 +1,23 @@
+interface Doer {
+	job string
+}
+
+fn (d Doer) do() string {
+	return '${d.job}'
+}
+
+struct Expector {
+	do_fn fn () string @[required]
+}
+
+struct Actioner {
+	job string
+}
+
+fn test_main() {
+	a := &Actioner{'dier'}
+	e := Expector{
+		do_fn: Doer(a).do
+	}
+	assert e.do_fn() == 'dier'
+}

